### ISSUE TITLE
eval (minikube -p minikube docker-env) doesn't work in fish-shell

### ISSUE THREAD

**Issue by agusti-t** [kubernetes/minikube#6595]
The command doesn't do what it's expected in fish-shell, may be related to: https://github.com/kubernetes/minikube/pull/6540
 
```
minikube version                                                                                                    Wed Feb 12 15:42:40 2020
minikube version: v1.7.2
commit: 50d543b5fcb0e1c0d7c27b1398a9a9790df09dfb
```

**The exact command to reproduce the issue**: `eval (minikube -p minikube docker-env)`

**The full output of the command that failed**:
```
 ~  minikube -p minikube docker-env                                                                                                                                              Wed Feb 12 16:30:00 2020
set -gx DOCKER_TLS_VERIFY "1"
set -gx DOCKER_HOST "tcp://192.168.64.33:2376"
set -gx DOCKER_CERT_PATH "/Users/agusti/.minikube/certs"
set -gx MINIKUBE_ACTIVE_DOCKERD "minikube"

# To point your shell to minikube's docker-daemon, run:
# eval (minikube -p minikube docker-env)
 ~  env | grep DOCKER                                                                                                                                                    328ms  Wed Feb 12 16:30:01 2020
 !  ~  eval (minikube -p minikube docker-env)                                                                                                                                   Wed Feb 12 16:30:15 2020
 ~  env | grep DOCKER                                                                                                                                                    192ms  Wed Feb 12 16:30:16 2020
DOCKER_TLS_VERIFY=1 set -gx DOCKER_HOST tcp://192.168.64.33:2376 set -gx DOCKER_CERT_PATH /Users/agusti/.minikube/certs set -gx MINIKUBE_ACTIVE_DOCKERD minikube
```

Quick workaround is to just copy and paste the set commands and run them manually:
```
set -gx DOCKER_TLS_VERIFY "1"
set -gx DOCKER_HOST "tcp://192.168.64.33:2376"
set -gx DOCKER_CERT_PATH "/Users/agusti/.minikube/certs"
set -gx MINIKUBE_ACTIVE_DOCKERD "minikube"
```

Something like:
```
 !  ~  set -e DOCKER_TLS_VERIFY                                                                                                                                                 Wed Feb 12 16:33:19 2020
 ~  env | grep DOCKER                                                                                                                                                            Wed Feb 12 16:33:36 2020
 !  ~  set -gx DOCKER_TLS_VERIFY "1"                                                                                                                                            Wed Feb 12 16:33:46 2020
         set -gx DOCKER_HOST "tcp://192.168.64.33:2376"
         set -gx DOCKER_CERT_PATH "/Users/agusti/.minikube/certs"
         set -gx MINIKUBE_ACTIVE_DOCKERD "minikube"
 !  ~  env | grep DOCKER                                                                                                                                                        Wed Feb 12 16:33:46 2020
DOCKER_CERT_PATH=/Users/agusti/.minikube/certs
DOCKER_HOST=tcp://192.168.64.33:2376
DOCKER_TLS_VERIFY=1
MINIKUBE_ACTIVE_DOCKERD=minikube
```

**The operating system version**: macOS 10.14.5

**Comment by afbjorklund** [comment]
Looks like both code and test was changed (semicolon removed), so looked intentional... @tstromberg ?

**Comment by afbjorklund** [comment]
Still broken, with `v1.7.3`

``` console
anders@anders-ThinkPad:~$ fish
Välkommen till fish, det vänliga interaktiva skalet
Skriv help för instruktioner om hur man använder fish
anders@anders-ThinkPad ~> minikube docker-env
export DOCKER_TLS_VERIFY="1"
export DOCKER_HOST="tcp://192.168.99.100:2376"
export DOCKER_CERT_PATH="/home/anders/.minikube/certs"
export MINIKUBE_ACTIVE_DOCKERD="minikube"

# To point your shell to minikube's docker-daemon, run:
# eval $(minikube -p minikube docker-env)
anders@anders-ThinkPad ~> eval $(minikube -p minikube docker-env)
$(...) is not supported. In fish, please use '(minikube)'.
fish: eval $(minikube -p minikube docker-env)
           ^
anders@anders-ThinkPad ~> fish --version
fish, version 2.2.0
anders@anders-ThinkPad ~> minikube version
minikube version: v1.7.3
commit: 436667c819c324e35d7e839f8116b968a2d0a3ff
```

Apparently it doesn't set SHELL, so that is why the detection fails.

But if you try, then it actually gets _worse_ (and definitely fails...)

```
anders@anders-ThinkPad ~> which fish
/usr/bin/fish
anders@anders-ThinkPad ~> export SHELL=/usr/bin/fish
anders@anders-ThinkPad ~> minikube docker-env
set -gx DOCKER_TLS_VERIFY "1"
set -gx DOCKER_HOST "tcp://192.168.99.100:2376"
set -gx DOCKER_CERT_PATH "/home/anders/.minikube/certs"
set -gx MINIKUBE_ACTIVE_DOCKERD "minikube"

# To point your shell to minikube's docker-daemon, run:
# eval (minikube -p minikube docker-env)
anders@anders-ThinkPad ~> eval (minikube -p minikube docker-env)
anders@anders-ThinkPad ~> env | grep DOCKER
DOCKER_TLS_VERIFY=1set-gxDOCKER_HOSTtcp://192.168.99.100:2376set-gxDOCKER_CERT_PATH/home/anders/.minikube/certsset-gxMINIKUBE_ACTIVE_DOCKERDminikube
```

Was this what you were seeing too, @kppullin ?

**Comment by kppullin** [comment]
@afbjorklund my output is similar, but without the `�` chars and instead plain space characters (more like @agusti-t's output).

My login shell is set to fish, in which case the `SHELL` var is properly initialized to `/usr/bin/fish` and the detection works in that case.  If I change the login shell to bash then I see the same behavior as your initial sample.

**Comment by afbjorklund** [comment]
Fixed in v1.8.0
